-1

**Failed Tests**:   Build  ClangBuild 
**Size**: This PR adds an extra 16KB to repository
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-fa37c4/43818/summary.html
**COMMIT**: c5b37af4ec78aec7636ec2a3b6cfaa0b114cfbd0
**CMSSW**: CMSSW_15_0_X_2025-01-16-2300/el8_amd64_gcc12
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week0/cms-sw/cmssw/47101/43818/install.sh` to create a dev area with all the needed externals and cmssw changes.



The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
 - @bsunanda cms-sw/cmssw#47112
 - @fabiocos cms-sw/cmssw#47115
 - @CMS-HGCAL cms-sw/cmssw#45665
 - @trackreco cms-sw/cmssw#47106
 - @Pruthvi-ch cms-sw/cmssw#47077
 - @cms-sw cms-sw/cmsdist#9623
 - @cms-sw cms-sw/cmsdist#9617
 - @cms-sw cms-sw/cmsdist#9621
 - @mseidel42 cms-sw/cmsdist#9615
 - @cms-sw cms-sw/cmsdist#9622

You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-fa37c4/43818/git-recent-commits.json
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-fa37c4/43818/git-merge-result


## Build

I found compilation error when building: 

<pre>>> Compiling edm plugin src/GeneratorInterface/Core/test/FailingGeneratorFilter.cc
/data/cmsbld/jenkins/workspace/ib-run-pr-tests/testBuildDir/el8_amd64_gcc12/external/gcc/12.3.1-40d504be6370b5a30e3947a6e575ca28/bin/c++ -c -DGNU_GCC -D_GNU_SOURCE -DTBB_USE_GLIBCXX_VERSION=120301 -DTBB_SUPPRESS_DEPRECATED_MESSAGES -DTBB_PREVIEW_RESUMABLE_TASKS=1 -DTBB_PREVIEW_TASK_GROUP_EXTENSIONS=1 -DBOOST_SPIRIT_THREADSAFE -DPHOENIX_THREADSAFE -DBOOST_MATH_DISABLE_STD_FPCLASSIFY -DBOOST_UUID_RANDOM_PROVIDER_FORCE_POSIX -DCMSSW_GIT_HASH='CMSSW_15_0_X_2025-01-16-2300' -DPROJECT_NAME='CMSSW' -DPROJECT_VERSION='CMSSW_15_0_X_2025-01-16-2300' -Isrc -Ipoison -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02872/el8_amd64_gcc12/cms/cmssw-patch/CMSSW_15_0_X_2025-01-16-2300/src -I/data/cmsbld/jenkins/workspace/ib-run-pr-tests/testBuildDir/el8_amd64_gcc12/external/highfive/2.3.1-b005372960e0853c83418fdf7b27ed23/include -I/data/cmsbld/jenkins/workspace/ib-run-pr-tests/testBuildDir/el8_amd64_gcc12/external/pcre/8.43-2d141998cfe5424b8f7aff48035cc2da/include -isystem/data/cmsbld/jenkins/workspace/ib-run-pr-tests/testBuildDir/el8_amd64_gcc12/external/boost/1.80.0-24621bb5019ee7e7984682e0f80b5fcf/include -I/data/cmsbld/jenkins/workspace/ib-run-pr-tests/testBuildDir/el8_amd64_gcc12/external/bz2lib/1.0.6-d065ccd79984efc6d4660f410e4c81de/include -isystem/data/cmsbld/jenkins/workspace/ib-run-pr-tests/testBuildDir/el8_amd64_gcc12/external/clhep/2.4.7.1-d3a3e353d370e701238f7949a0d7909f/include -I/data/cmsbld/jenkins/workspace/ib-run-pr-tests/testBuildDir/el8_amd64_gcc12/external/fastjet/3.4.1-43adababf6036a99696d8ed78b995b5c/include -I/data/cmsbld/jenkins/workspace/ib-run-pr-tests/testBuildDir/el8_amd64_gcc12/external/gsl/2.6-f7574c606b0ce57ff601d3ca9534cd01/include -I/data/cmsbld/jenkins/workspace/ib-run-pr-tests/testBuildDir/el8_amd64_gcc12/external/hdf5/1.14.1-5dbe775e7ef35da13f2aee909660c0aa/include -I/data/cmsbld/jenkins/workspace/ib-run-pr-tests/testBuildDir/el8_amd64_gcc12/external/hepmc/2.06.10-262d73524c32528e87adc31a98f68d52/include -I/data/cmsbld/jenkins/workspace/ib-run-pr-tests/testBuildDir/el8_amd64_gcc12/external/heppdt/3.04.01-bf3e105f1c3c5d5f64e00d5c18e71fb2/include -I/data/cmsbld/jenkins/workspace/ib-run-pr-tests/testBuildDir/el8_amd64_gcc12/external/lhapdf/6.4.0-a73374d8d6da1aa2194ba51da76db01b/include -I/data/cmsbld/jenkins/workspace/ib-run-pr-tests/testBuildDir/el8_amd64_gcc12/external/libuuid/2.34-27ce4c3579b5b1de2808ea9c4cd8ed29/include -isystem/data/cmsbld/jenkins/workspace/ib-run-pr-tests/testBuildDir/el8_amd64_gcc12/lcg/root/6.32.09-2f25210d9b2d495c49b4a9094cb275ed/include -isystem/data/cmsbld/jenkins/workspace/ib-run-pr-tests/testBuildDir/el8_amd64_gcc12/external/tbb/v2021.9.0-573155027234b8f945d29403a2749d52/include -I/data/cmsbld/jenkins/workspace/ib-run-pr-tests/testBuildDir/el8_amd64_gcc12/external/xerces-c/3.1.3-c7b88eaa36d0408120f3c29826a04bf6/include -I/data/cmsbld/jenkins/workspace/ib-run-pr-tests/testBuildDir/el8_amd64_gcc12/external/xz/5.2.5-6f3f49b07db84e10c9be594a1176c114/include -I/data/cmsbld/jenkins/workspace/ib-run-pr-tests/testBuildDir/el8_amd64_gcc12/external/zlib/1.2.13-d217cdbdd8d586e845e05946de2796be/include -I/data/cmsbld/jenkins/workspace/ib-run-pr-tests/testBuildDir/el8_amd64_gcc12/external/fmt/10.2.1-e35fd1db5eb3abc8ac0452e8ee427196/include -I/data/cmsbld/jenkins/workspace/ib-run-pr-tests/testBuildDir/el8_amd64_gcc12/external/hepmc3/3.2.7-f10dce5b16f0246f8cac8378933b9681/include -I/data/cmsbld/jenkins/workspace/ib-run-pr-tests/testBuildDir/el8_amd64_gcc12/external/md5/1.0.0-5b594b264e04ae51e893b1d69a797ec6/include -I/data/cmsbld/jenkins/workspace/ib-run-pr-tests/testBuildDir/el8_amd64_gcc12/external/OpenBLAS/0.3.27-70a9dd2c9f309171934f13e3003b0540/include -I/data/cmsbld/jenkins/workspace/ib-run-pr-tests/testBuildDir/el8_amd64_gcc12/external/openmpi/4.1.6-e8311880fc338dbd65c6aa8779ca36f5/include -I/data/cmsbld/jenkins/workspace/ib-run-pr-tests/testBuildDir/el8_amd64_gcc12/external/tinyxml2/6.2.0-f99ae2781d074227d47e8a3e7c8ec87e/include -O3 -pthread -pipe -Werror=main -Werror=pointer-arith -Werror=overlength-strings -Wno-vla -Werror=overflow -std=c++20 -ftree-vectorize -Werror=array-bounds -Werror=format-contains-nul -Werror=type-limits -fvisibility-inlines-hidden -fno-math-errno --param vect-max-version-for-alias-checks=50 -Xassembler --compress-debug-sections -Wno-error=array-bounds -Warray-bounds -fuse-ld=bfd -march=x86-64-v3 -felide-constructors -fmessage-length=0 -Wall -Wno-non-template-friend -Wno-long-long -Wreturn-type -Wextra -Wpessimizing-move -Wclass-memaccess -Wno-cast-function-type -Wno-unused-but-set-parameter -Wno-ignored-qualifiers -Wno-unused-parameter -Wunused -Wparentheses -Werror=return-type -Werror=missing-braces -Werror=unused-value -Werror=unused-label -Werror=address -Werror=format -Werror=sign-compare -Werror=write-strings -Werror=delete-non-virtual-dtor -Werror=strict-aliasing -Werror=narrowing -Werror=unused-but-set-variable -Werror=reorder -Werror=unused-variable -Werror=conversion-null -Werror=return-local-addr -Wnon-virtual-dtor -Werror=switch -fdiagnostics-show-option -Wno-unused-local-typedefs -Wno-attributes -Wno-psabi -Wno-error=unused-variable -DBOOST_DISABLE_ASSERTS -flto=auto -fipa-icf -flto-odr-type-merging -fno-fat-lto-objects -Wodr -fPIC -MMD -MF tmp/el8_amd64_gcc12/src/GeneratorInterface/Core/test/FailingGeneratorFilter/FailingGeneratorFilter.cc.d src/GeneratorInterface/Core/test/FailingGeneratorFilter.cc -o tmp/el8_amd64_gcc12/src/GeneratorInterface/Core/test/FailingGeneratorFilter/FailingGeneratorFilter.cc.o
In file included from src/GeneratorInterface/Core/test/FailingGeneratorFilter.cc:17:
src/GeneratorInterface/Core/interface/GeneratorFilter.h: In instantiation of 'bool edm::GeneratorFilter<HAD, DEC>::filter(edm::Event&, const edm::EventSetup&) [with HAD = test::FailingHad; DEC = test::DummyDec]':
src/GeneratorInterface/Core/interface/GeneratorFilter.h:149:8:   required from here
src/GeneratorInterface/Core/interface/GeneratorFilter.h:194:46: error: cannot convert 'std::unique_ptr<HepMC3::GenEvent>::pointer' {aka 'HepMC3::GenEvent*'} to 'const HepMC::GenEvent*'
  194 |           auto t = decayer_->decay(event3.get());
      |                                    ~~~~~~~~~~^~
      |                                              |
      |                                              std::unique_ptr<HepMC3::GenEvent>::pointer {aka HepMC3::GenEvent*}
src/GeneratorInterface/Core/test/FailingGeneratorFilter.cc:121:28: note:   initializing argument 1 of 'HepMC::GenEvent* test::DummyDec::decay(const HepMC::GenEvent*)'
</pre>

## Clang Build

I found compilation error while trying to compile with clang. Command used:
```
USER_CUDA_FLAGS='--expt-relaxed-constexpr' USER_CXXFLAGS='-Wno-register -fsyntax-only' /usr/bin/time -v scram build -k -j 32 COMPILER='llvm compile'

```


<pre>>> Entering Package SimTracker/SiStripDigitizer
>> Entering Package SimTracker/TrackTriggerAssociation
>> Entering Package TauAnalysis/MCEmbeddingTools
>> Entering Package Utilities/ReleaseScripts
>> Compile sequence completed for CMSSW CMSSW_15_0_X_2025-01-16-2300
gmake: *** [There are compilation/build errors. Please see the detail log above.] Error 1
Command exited with non-zero status 1
	Command being timed: "scram build -k -j 32 COMPILER=llvm compile BUILD_LOG=yes"
	User time (seconds): 27186.33
	System time (seconds): 3247.75
	Percent of CPU this job got: 1342%
</pre>

--- a/GeneratorInterface/Core/test/FailingGeneratorFilter.cc
+++ b/GeneratorInterface/Core/test/FailingGeneratorFilter.cc
@@ -119,6 +119,7 @@ namespace test {
     std::vector<std::string> sharedResources() const { return {}; }
 
     HepMC::GenEvent* decay(HepMC::GenEvent const*) { return nullptr; }
+    HepMC3::GenEvent* decay(HepMC3::GenEvent const*) { return nullptr; }
     void statistics() const {}
 
     void init(const edm::EventSetup&) const {}

